enhancements/machine-api/control-plane-set.md

```diff
@@ -0,0 +1,603 @@
+reduce the likelihood of datacenter level faults degrading the control plane.
+To this end, the failure domains for the control plane will be set on the ControlPlaneSet spec directly.
+
+When creating Machines, the ControlPlaneSet controller will balance the Machines across these failure domains by
+injecting the desired failure domain for the new Machine into the provider spec based on the platform specific failure
+domain field.
+
+#### Ensuring Machines match the desired state
+
+To ensure Machines are up to date with the desired configuration, the ControlPlaneSet controller will leverage a
+similar pattern to the workload controllers.
+It will hash the template and compare the hashed result with the spec of the Machines within the managed set.
+As we expect the failure domain to vary across the managed Machines, this will need to be omitted before the hash can
+be calculated.
+
+Should any Machine not match the desired hash, it will be updated based on the chosen update strategy.
+
+##### The RollingUpdate update strategy
+
+The RollingUpdate strategy will be the default. It mirrors the RollingUpdate strategy familiar to users from
+Deployments. When a change is required to be rolled out, the Machine controller will create a new Control Plane
+Machine, wait for this to become ready, and then remove the old Machine. It will repeat this process until all Machines
+are up to date.
+
+At first, we will not allow any configuration of the RollingUpdate (as you might see in a Deployment) and will pin the
+surge to 1 replica. We may wish to change this later once we have more data about the stability of this operator and
+the etcd protection mechanism it relies on.
+
+We expect this strategy to be used in most applications of the ControlPlaneSet, in particular it can only not be used
+in environments where capacity is very limited and a surge of any control plane capacity is unavailable.
+
+##### The Recreate update strategy
+
+The Recreate strategy mirrors the Recreate strategy familiar to users from Deployments.
+When a change is required to be rolled out, the Machine controller will first remove a Control Plane Machine, wait for
+its removal, and then create a new Control Plane Machine.
+
+This strategy is intended to be used only in very resource constrained environments where there is no capacity to
+introduce an extra temporary Control Plane Machine (preventing the RollingUpdate strategy).
+
+At present, when using this strategy, the updates will need manual intervention. The etcd protection design will
+prevent the Control Plane Machine from being removed until a replacement for the etcd member has been introduced into
+the cluster. To allow for this use case, the end user can manually remove the etcd protection (via removing the Machine
+deletion hook on the deleted Machine) allowing the rollout to proceed, the etcd operator will not re-add the
+protection if the Machine is already marked for deletion.
```

Comment:
The operator will see that the Machine is not being drained (machines have a condition for this), it can report that back in its own status, machine foo isn't being drained because etcd is blocking it.

Then, when the etcd operator pre-drain hook is removed by the end user, this will go away, the drain will occur and the Machine goes away. The ControlPlaneSet will then spot the Machine has gone, and create the replacement.